src/library/documentation.cpp

```diff
@@ -36,15 +36,25 @@ static environment update(environment const & env, documentation_ext const & ext
 struct doc_modification : public modification {
     LEAN_MODIFICATION("doc")
 
+    /** If non-empty, this is a doc on a declaration. Otherwise,
+      * it's a doc on the whole module. */
     name m_decl;
```

Comment:
I tried to use `optional<name>` here, where module docs would have the field empty and declaration docs have a proper name here, but this seems to break .olean ouput, probably in the `doc_modification::serialize` function.